src/stapi_fastapi/backends/root_backend.py

```diff
@@ -6,15 +6,17 @@
 
 from stapi_fastapi.models.order import (
     Order,
-    OrderCollection,
     OrderStatus,
 )
 
 
 class RootBackend[T: OrderStatus](Protocol):  # pragma: nocover
-    async def get_orders(self, request: Request) -> ResultE[OrderCollection]:
+    async def get_orders(
+        self, request: Request, next: str | None, limit: int
+    ) -> ResultE[tuple[list[Order], str]]:
```

Comment:
I think this would be better typed with:

```suggestion
    ) -> ResultE[tuple[list[Order], Maybe[str]]]:
```

Where the end of pages is signaled with `Nothing` and a pagination token is signaled with `Some(s)`. I think using the empty string as the "no pagination" value is not a good practice.

Reply:
changed to suggested approach